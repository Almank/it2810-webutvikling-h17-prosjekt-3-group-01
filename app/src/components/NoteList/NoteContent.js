/**
 * Created by martinlunde on 27.09.2017.
 */
import React from 'react';
import { Text, View, StyleSheet, TextInput } from 'react-native';
import { Button } from 'react-native-elements';

export class NoteContent extends React.Component {
    constructor(props){
        super(props);
        this.state = {
            textfield: "",
        };
        this.handleChange = this.handleChange.bind(this);
        this.returnToLastView = this.returnToLastView.bind(this);
        this.handleDelete = this.handleDelete.bind(this);
    }

    returnToLastView(){
        this.props.navigation.goBack();
    }

    handleChange(text){
        let data = this.state;
        data.textfield = text;
        this.setState(data);
    }

    handleDelete(){
        this.props.navigation.state.params.onPress();
        this.returnToLastView();
    }

    render(){
        return(
            <View>
                <View style={styles.header}>
                    <Button buttonStyle={styles.headerButton}
                            title="< Back"
                            fontWeight={"bold"}
                            onPress={this.returnToLastView}/>
                </View>
                <View style={styles.noteContainer}>
                    <View style={styles.hr}/>
                    <Text style={styles.headerTitle}>{this.props.navigation.state.params.title}</Text>
                    <View style={styles.hr}/>
                    <TextInput style={styles.noteField}
                               onChangeText={(text) => this.handleChange(text)}
                               value={this.state.text}
                               multiline={true}
                               numberOfLines={4}
                    />
                    <Button title="Delete Note"
                            fontWeight={'bold'}
                            fontSize={15}
                            onPress={this.handleDelete}
                            buttonStyle={styles.DeleteButton}
                    />
                </View>
            </View>
        );
    }
}

const styles = StyleSheet.create({
    header: {
        display: 'flex',
        flexDirection: 'row',
        alignItems: 'center',
        backgroundColor: 'white',
    },
    headerButton: {
        marginTop: 5,
        marginBottom: 5,
        marginLeft: -10,
        backgroundColor: '#35393a',
    },
    headerTitle: {
        textAlign: 'center',
        marginTop: 5,
        marginBottom: 5,
        fontSize: 15,
        fontWeight: 'bold',
    },
    hr: {
        borderBottomColor: 'grey',
        borderBottomWidth: 1,
    },
    noteContainer: {
        display: 'flex',
    },
    noteField: {
        backgroundColor: 'white',
        height: 300,
    },
    DeleteButton: {
        marginTop: 10,
        backgroundColor: '#8f0500',
    }
});