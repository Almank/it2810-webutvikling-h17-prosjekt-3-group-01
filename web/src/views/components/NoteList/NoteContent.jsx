/**
 * Created by martinlunde on 27.09.2017.
 */
import React from 'react';

export class NoteContent extends React.Component {
    constructor(props){
        super(props);
        this.state = {
            title: this.props.data.title,
            content: this.props.data.content
        };

        this.handleChange = this.handleChange.bind(this);
    }

    handleChange(event){
        this.setState({content: event.target.value});
        this.props.onChange(this.state.title, event.target.value);
    }

    componentWillReceiveProps(nextProps){
        this.setState({
            title: nextProps.data.title,
            content: nextProps.data.content
        });
    }

    render(){
        //Handles empty boxes, so that previous states not are stuck.
        let textValue;
        if(!this.state.content){
            textValue = '';
        } else {
            textValue = this.state.content;
        }

        return(
            <div className="NoteContentContainer">
                <h1>{ this.state.title }</h1>
                <textarea value={ textValue } onChange={this.handleChange} />
            </div>
        );
    }
}