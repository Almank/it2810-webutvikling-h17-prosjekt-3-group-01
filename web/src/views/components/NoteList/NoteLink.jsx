/**
 * Created by martinlunde on 27.09.2017.
 */
import React from 'react';
import propTypes from 'prop-types';
import {Link} from 'react-router-dom';

//The class NoteLink renders a note-link-object for the parent NoteList.
export const NoteLink = ({index, title, onClick}) => {
    return(
        <div className="NoteLinkContainer">
            <Link className="NoteButton"
                  key={index}
                  to={`/notebook/${title}`}>

                <h1>{title}</h1>
            </Link>
            <button className="RemoveButton"
                    type="button"
                    onClick={onClick}
                    value={title}>
                <span className='glyphicon glyphicon-minus' />
            </button>
        </div>
    );
};


NoteLink.propTypes = {
    title: propTypes.string,
};

NoteLink.defaultProps = {
    title: 'Note',
};