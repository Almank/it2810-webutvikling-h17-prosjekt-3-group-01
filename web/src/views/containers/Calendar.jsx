/**
 * Created by almank on 02.10.2017.
 */
import React from 'react';
import {WeekCal, AppointmentForm, ContentHeader} from '../Components';
import '../../assets/styles/Calendar.css';

export class Calendar extends React.Component {
    constructor(props) {

        //Setting state if there is none stored.
        if(localStorage.getItem("emptyCalendar") === null){
            let data = {
                dateToday: new Date().toISOString().slice(0, 10),
                children: [],
            };
            localStorage.setItem("emptyCalendar", JSON.stringify(data));
        }
        super(props);

        //Fetching state from storage.
        let data = localStorage.getItem("emptyCalendar");
        data = JSON.parse(data);

        //Setting dateToday to current date.
        data.dateToday = new Date().toISOString().slice(0, 10);

        //Setting state to stored state.
        this.state = data;

        //Binding functions
        this.createAppointment = this.createAppointment.bind(this);
        this.changeContent = this.changeContent.bind(this);
        this.handleRemoveClick = this.handleRemoveClick.bind(this);
        this.updateLocalStorage = this.handleRemoveClick.bind(this);
        this.handleChangeClick = this.handleChangeClick.bind(this);
    }

    handleRemoveClick(event){
        for (let i=0; i<this.state.children.length; i++) {
            if (String(this.state.children[i].uniqueDate) === String(event.target.value)){
                //Deleting values in child
                delete this.state.children[i];

                //Removing empty objects
                let newArr = this.state.children.filter(val => Object.keys(val).length !== 0);

                //Updating state and localstorage on with callback function.
                this.setState({children: newArr}, function() {
                    localStorage.setItem("emptyCalendar", JSON.stringify(this.state));
                });
            }
        }
    }

    handleChangeClick(type, event){
        for (let i=0; i<this.state.children.length; i++) {
            if (String(this.state.children[i].uniqueDate) === String(event.target.name)) {
                let data = this.state.children;
                if (type === 'title')
                    data[i].title = event.target.value;
                else if (type === 'time')
                    data[i].time = event.target.value;
                else if (type === 'text')
                    data[i].text = event.target.value;
                this.setState({children: data});
                localStorage.setItem("emptyCalendar", JSON.stringify(this.state));
            }
        }
    }

    emptyScheduleCheck(){
        //Copy of the state children array.
        let myData =[].concat(this.state.children)

            //Filtering out appointments on current day.
            .filter(child => child.date === this.state.dateToday)

            //Sorting elements based on time, earliest first.
            .sort((a, b) => parseInt((("" + a.time.slice(0,2)) + a.time.slice(3,6)), 0) -
                            parseInt(("" + b.time.slice(0,2)) + b.time.slice(3,6), 0))

            //Mapping items from array giving the html the correct values.
            .map((item,i) =>
            <div className='calendarBoxContent' key={i}>
                <div className='leftCalendarBoxContent'>
                    <input onBlur={this.handleChangeClick.bind(this, 'time')}
                           type='time'
                           defaultValue={this.state.children.filter(a => a.uniqueDate === item.uniqueDate)[0].time}
                           name={item.uniqueDate}
                           className='changeField'/>
                </div>

                <div className='middleCalendarBoxContent'>
                    <input onBlur={this.handleChangeClick.bind(this, 'title')}
                           type='text'
                           defaultValue={this.state.children.filter(a => a.uniqueDate === item.uniqueDate)[0].title}
                           name={item.uniqueDate}
                           maxLength={10}
                           className='changeField'/>
                </div>
                <div className='rightCalendarBoxContent'>
                    <textarea onBlur={this.handleChangeClick.bind(this, 'text')}
                           type='text'
                           defaultValue={this.state.children.filter(a => a.uniqueDate === item.uniqueDate)[0].text}
                           name={item.uniqueDate}
                           maxLength={200}
                           className='changeField'/>
                </div>
                <button onClick={this.handleRemoveClick}
                        className='RemoveButton'
                        type='button'
                        value={item.uniqueDate}>
                    <span className='glyphicon glyphicon-minus' />
                </button>
            </div>
            );
        //If there are no plans that day.
        if (myData.length === 0){
            myData = [  <div key={0} className='calendarBoxContent'>
                            <h1>You don't have any plans!</h1>
                        </div>];
            }

        return (
            <div className='appointmentFlow'>
                {myData}
            </div>
        );
    }

    createAppointment(e) {
        //Preventing reload of window.
        e.preventDefault();

        //Fetching values from form.
        let dateValue = document.getElementsByClassName('dateInput')[0].value;
        let timeValue = document.getElementsByClassName('timeInput')[0].value;
        let titleValue = document.getElementsByClassName('titleInput')[0].value;
        let textValue = document.getElementsByClassName('textInput')[0].value;

        //Validate fields date, time and title. Last field is optional.
        if (this.validateFormDate(dateValue) && this.validateFormTime(timeValue) && this.validateFormTitle(titleValue)){

            //Creating new array with current state children.
            let newStateArray = this.state.children.slice();

            //Pushing new child to array.
            newStateArray.push({date: dateValue, time: timeValue, title: titleValue, text: textValue, uniqueDate: new Date()});

            //Sorting array with earliest appointments first.
            newStateArray.sort((a, b) =>    parseInt(("" + a.time.slice(0,2)) + a.time.slice(3,6), 0) -
                                            parseInt(("" + b.time.slice(0,2)) + b.time.slice(3,6), 0));

            //Creating new updated state.
            let data = {
                children: newStateArray,
                dateToday: new Date().toISOString().slice(0, 10),
            };

            //Setting state.
            this.setState(data);

            //Updating localstorage to store new data.
            localStorage.setItem("emptyCalendar", JSON.stringify(data));

            //Resetting form values.
            document.getElementsByClassName('dateInput')[0].value = null;
            document.getElementsByClassName('timeInput')[0].value = null;
            document.getElementsByClassName('titleInput')[0].value = null;
            document.getElementsByClassName('textInput')[0].value = null;

            //Hiding form.
            this.showForm();
        } else {
            alert('Invalid values. Please try again.')
        }
    }

    validateFormDate(date){
        return date.length === 10 && date >= new Date().toISOString().slice(0, 10);
    }

    validateFormTime(time){
        return time.length === 5;
    }

    validateFormTitle(title){
        return title.length > 0;
    }

    changeContent(e){
        //Fetching data from localstorage
        let data = localStorage.getItem("emptyCalendar");
        data = JSON.parse(data);

        //Setting dateToday to current date.
        data.dateToday = e.props.dateFull;

        //Setting state to stored state.
        this.state = data;

        //Updating localstorage to store new data.
        localStorage.setItem("emptyCalendar", JSON.stringify(data));

        //Forcing update of the DOM to change content.
        this.forceUpdate();
    }

    showForm(){
        //Toggle form.
        let element = document.querySelector('.formContainer');
        element.style.display = element.style.display === 'none' ? 'flex' : 'none';
        let today = new Date().toISOString().slice(0, 10);
        document.querySelector(".dateInput").value = today;
        let fields = document.querySelectorAll(".inputField");
        for (let i = 0; i < fields.length; i++)
            fields[i].style.border = '1px solid lightgray';

    }

    render(){
        return (
            <div className={'calendarContent'}>
                <h1 className={'title'} >Your schedule for the coming week</h1>
                <WeekCal change={this.changeContent} />

                <div className={'calendarBox'}>
                    <ContentHeader closeForm={this.showForm}/>
                    <AppointmentForm closeForm={this.showForm} submitForm={this.createAppointment}/>
                    {this.emptyScheduleCheck()}
                </div>
            </div>
        );
    }
}