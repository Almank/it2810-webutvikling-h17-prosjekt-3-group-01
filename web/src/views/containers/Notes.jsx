/**
 * Created by martinlunde on 27.09.2017.
 */
import React from 'react';
import '../../assets/styles/Notes.css';
import {NoteList, NoteContent} from "../Components";
import {Route} from 'react-router-dom';


export class Notes extends React.Component {
    constructor(props){
        //Initiate local storage if no data is stored.
        if(localStorage.getItem("notes") === null){
            let data = {
                notes: {
                    'default': {
                        title: 'default',
                        content: 'no content'
                    }
                },
                update: false
            };
            localStorage.setItem("notes", JSON.stringify(data));
        }

        //Load locally stored data into state.
        let notes = localStorage.getItem("notes");
        notes = JSON.parse(notes);
        super(props);
        this.state = {
            notes: notes['notes'],
        };
        this.removeClick = this.removeClick.bind(this);
    };

    //Function for updating locally stored data.
    updateLocalStorage(){
        let data = this.state;
        localStorage.setItem("notes", JSON.stringify(data));
    }

    //Function for updating a notes title and content.
    onChange(title, content){
        // eslint-disable-next-line
        this.state.notes[title].title = title;
        // eslint-disable-next-line
        this.state.notes[title].content = content;
        this.updateLocalStorage();
    }

    //Function for enabling remove button and updating storage.
    removeClick(title){
        delete this.state.notes[title];
        this.updateLocalStorage();
        this.setState({update: true});
        window.location.pathname = "/notebook";
    }

    render(){
        return (
            <div className="NoteBookWrapper">
                <h1 className={'title'}> Notes</h1>
                <div className="NoteBook">
                    <NoteList notes={this.state.notes}
                              removeClick={(value) => this.removeClick(value)}
                              updateStorage={() => this.updateLocalStorage()}
                    />
                    <div className="NoteEdit">
                        <div className="NoteEditContent">
                            <Route exact path="/notebook/:name" render={ (id) =>
                                <NoteContent data={this.state.notes[id.match.params.name]}
                                             key={id}
                                             id={id}
                                             onChange={this.onChange.bind(this)}
                                             update={this.state.update}
                                />}/>
                        </div>
                    </div>
                </div>
            </div>
        );
    }
}