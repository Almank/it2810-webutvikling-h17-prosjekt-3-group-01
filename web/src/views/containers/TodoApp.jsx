import React from 'react';
import '../../assets/styles/Todolists.css';
import {TodoForm, TodoList, TodoFilter, TodoCatelog, TodoCatelogForm} from '../Components';

/* [TODO APP] */
export class ToDoApp extends React.Component{
	constructor(props){
		super(props);

		if(localStorage.getItem("todo") === null){
			let data = {Todo: [],
								selectedItem: "0"};
			localStorage.setItem("todo", JSON.stringify(data));
		}
		let data = {Todo: [{name:"Primary",
			items:[]},],
			filter: [{keyword:'',Status:"SHOW_ALL"}],
			selectedItem: "0"};
		localStorage.setItem("todo", JSON.stringify(data));
		let todo = localStorage.getItem("todo");
		todo = JSON.parse(todo);
		this.state = todo;

		this.updateItems = this.updateItems.bind(this);
		this.deleteItem = this.deleteItem.bind(this);
		this.filterItem = this.filterItem.bind(this);
		this.AddCatalog = this.AddCatalog.bind(this);
		this.searchItem = this.searchItem.bind(this);
		this.filterItem = this.filterItem.bind(this);
		this.setSelectedCatalog = this.setSelectedCatalog.bind(this);
	}

	updateItems(newItem){
		let item = {item:newItem,isDone:false};
		let newtodo = this.state;
		let allItems = this.state.Todo[this.state.selectedItem].items.concat([item]);
		newtodo.Todo[this.state.selectedItem].items = allItems;
		this.setState(newtodo);
		localStorage.setItem("todo", JSON.stringify(this.state));
	}

	deleteItem(index){
		let newtodo = this.state.Todo;
		let allItems = this.state.Todo[this.state.selectedItem].items.slice(); //copy array
		allItems.splice(index, 1); //remove element
		newtodo[this.state.selectedItem].items = allItems;
		this.setState(newtodo);
		localStorage.setItem("todo", JSON.stringify(this.state));
	}

	filterItem(e){
		let filter = this.state.filter;
		filter[0].Status = e.target.id;
		this.setState({
			filter: filter,
		});
	}

	searchItem(e){
		let filter = this.state.filter;
		filter[0].keyword = e.target.value;
		this.setState({
			filter: filter,
		});
	}

	AddCatalog(newCatalog){
		let Catalog = {name:newCatalog,items:[{item:'Todo item #1',isDone:false}]};
		let newtodo = this.state.Todo.concat([Catalog]);
		this.setState({Todo: newtodo});
		localStorage.setItem("todo", JSON.stringify(this.state));
	}

	setSelectedCatalog(index){
		this.setState({
			selectedItem: index,
		});
	}

	render(){
		return (
			<div className="todo-wrapper">
				<h1 className={'title'} >To-do list</h1>
				<div className="bigrow">
					<div className="col1">
						<h3>To-do lists:</h3>
						<TodoCatelogForm onFormSubmit = {this.AddCatalog} />
						<TodoCatelog selectedID={this.state.selectedItem} onClick={this.setSelectedCatalog} Todos = {this.state.Todo} />
					</div>
					<div className="col2">
						<h3>What do you want to-do?</h3>
						<TodoFilter onFilter = {this.filterItem} onSearch = {this.searchItem} filter={this.state.filter}/>
						<TodoForm onFormSubmit = {this.updateItems} />
						<TodoList  items = {this.state.Todo[this.state.selectedItem].items} filter = {this.state.filter} onDelete={this.deleteItem}/>
					</div>
				</div>
			</div>
		);
	}

}