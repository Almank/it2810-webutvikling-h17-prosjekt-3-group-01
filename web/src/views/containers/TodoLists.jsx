/**
 * Created by martinlunde on 27.09.2017.
 */
import React from 'react';
import ToDoApp from "../components/TodoList/TodoApp";


export class TodoLists extends React.Component {
    render(){
        return ( 
            <div id="todolist">
                <ToDoApp/>
            </div>

        );


    }
}