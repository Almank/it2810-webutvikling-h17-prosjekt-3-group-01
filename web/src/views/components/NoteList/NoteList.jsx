/**
 * Created by martinlunde on 27.09.2017.
 */
import React from 'react';
import '../../../assets/styles/Notes.css';
import {NoteLink} from "./NoteLink";

//The class NoteList is the parent of all notes in the list of nodes.
export class NoteList extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            notes: this.props.notes,
        };

        this.onSubmit = this.onSubmit.bind(this);
        this.handleRemoveClick = this.handleRemoveClick.bind(this);
    }

    //Function for adding new note and updating storage to contain this.
    onSubmit(event){
        if(event.target.title.value !== "") {
            const data = this.state.notes;
            data[event.target.title.value] = {
                title: event.target.title.value,
            };
            this.setState({notes: data});
            this.props.updateStorage();
        }
        event.preventDefault();
    }

    //Function for handling the removeclick and has to force update due to the state of its parent.
    handleRemoveClick(event){
        let title = event.target.value;
        this.props.removeClick(title);
        this.forceUpdate();
    }

    //Function for rendering all the listed notes. Uses forloop because of dictionaries.
    renderListedNotes(){
        let dataSet = [];
        const data = this.state.notes;
        let index = 0;
        for(let key in data){
            dataSet.push(<NoteLink title={ data[key].title }
                                   content={data[key].content}
                                   key={index}
                                   onClick={this.handleRemoveClick}
            />);
            index++;
        }
        return dataSet;
    }

    render(){
        return (
            <div className="NoteList">
                <div className="NoteListTitleContainer">
                    <h1>Add new</h1>
                </div>
                <form className="NoteForm" onSubmit={this.onSubmit}>
                    <input className="NoteField" name="title" type="text"/>
                    <input className="NoteAdd" type="submit" value="ADD"/>
                </form>
                <div className="CurrentNotes">
                    { this.renderListedNotes() }
                </div>
            </div>
        );
    }
}
