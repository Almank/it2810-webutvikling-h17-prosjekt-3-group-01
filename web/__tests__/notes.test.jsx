import React from 'react';
import localStorage from "../__test_setup__/localstorage";

import {Notes} from '../src/views/containers/Notes';
import {NoteList} from "../src/views/components/NoteList/NoteList";
import {NoteLink} from "../src/views/components/NoteList/NoteLink";
import {NoteContent} from "../src/views/components/NoteList/NoteContent";


describe("Notes", ()=>{
    it("Should render correctly", () =>{
       const SHALLOW = shallow(<Notes/>);
       expect(SHALLOW).toMatchSnapshot();
    });

    describe("NoteList", () => {
        it("Should render correctly", () => {
            const SHALLOW = shallow(<NoteList/>);
            expect(SHALLOW).toMatchSnapshot();
        });
    });

    describe("NoteLink", () => {
        it("Should render correctly", () => {
            const SHALLOW = shallow(<NoteLink/>);
            expect(SHALLOW).toMatchSnapshot();
        });
    });


    describe("NoteContent", () => {
        it("Should render correctly", () =>{
            let data = {
                'default': {
                    title: 'default',
                    content: 'no content'
                }
            };
            const SHALLOW = render(<NoteContent data={data}/>);
            expect(SHALLOW).toMatchSnapshot();
        });
    });

});