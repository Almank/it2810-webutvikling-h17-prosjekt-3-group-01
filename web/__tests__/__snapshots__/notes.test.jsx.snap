// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notes NoteContent Should render correctly 1`] = `
<div
  class="NoteContentContainer"
>
  <h1 />
  <textarea />
</div>
`;

exports[`Notes NoteLink Should render correctly 1`] = `
<div
  className="NoteLinkContainer"
>
  <Link
    className="NoteButton"
    replace={false}
    to="/notebook/Note"
  >
    <h1>
      Note
    </h1>
  </Link>
  <button
    className="RemoveButton"
    type="button"
    value="Note"
  >
    <span
      className="glyphicon glyphicon-minus"
    />
  </button>
</div>
`;

exports[`Notes NoteList Should render correctly 1`] = `
<div
  className="NoteList"
>
  <div
    className="NoteListTitleContainer"
  >
    <h1>
      Add new
    </h1>
    <div
      className="addNewNoteButton"
      onClick={[Function]}
    >
      <span
        className="glyphicon glyphicon-plus"
      />
    </div>
  </div>
  <form
    className="NoteForm"
    onSubmit={[Function]}
  >
    <input
      className="NoteField"
      name="title"
      type="text"
    />
    <input
      className="NoteAdd"
      type="submit"
      value="ADD"
    />
  </form>
  <div
    className="CurrentNotes"
  />
</div>
`;

exports[`Notes Should render correctly 1`] = `
<div>
   
  <h1
    className="title"
  >
    Notes
  </h1>
  <div
    className="NoteBook"
  >
    <NoteList
      notes={
        Object {
          "default": Object {
            "content": "no content",
            "title": "default",
          },
        }
      }
      removeClick={[Function]}
      updateStorage={[Function]}
    />
    <div
      className="NoteEdit"
    >
      <div
        className="NoteEditContent"
      >
        <Route
          exact={true}
          path="/notebook/:name"
          render={[Function]}
        />
      </div>
    </div>
  </div>
</div>
`;
